internal/polkavm/interpreter/instance.go

```diff
@@ -50,11 +50,11 @@ func (i *Instance) skip() {
 	i.instructionCounter += 1 + polkavm.Skip(i.instructionCounter, i.bitmask)
 }
 
-func (i *Instance) deductGas(cost polkavm.Gas) error {
-	if i.gasRemaining < int64(cost) {
+func (i *Instance) deductGas() error {
+	if i.gasRemaining <= 0 {
 		return polkavm.ErrOutOfGas
 	}
-	i.gasRemaining -= int64(cost)
+	i.gasRemaining--
```

Comment:
this will work for now since the gas cost is 1 but it is not GP compliant. This subtraction needs to be for each instruction, which unfortunately will require a bigger refactor